Docs: Filtering Metrics in Surfacers

robpickerill opened the pull request:
Staging this PR to align filtering docs as a follow on for #597. I'd spotted the cloudwatch docs needed a quick update so I took the opportunity to also write a bit of documentation for filtering metrics.

I'd also spotted I'd missed the surfacers overview -> cloudwatch reference so added that here.

There is a minor code change in the protobuf config and tests for cloudwatch so I think this PR needs splitting again? Let me know what's best.

manugarg commented on docs/content/surfacers/overview.md:
```diff
@@ -48,3 +49,66 @@ surfacer {
+
+#### Filtering by Label
+
+Cloudprober can filter the metrics that are published to surfacers. To filter metric labels, reference one of the following keys in the surfacer configuration:
```
s/To filter metric labels/To filter metrics by labels/

robpickerill replied:
Thanks, fixed here: https://github.com/google/cloudprober/pull/598/commits/4a4c5e982badc7944488fa1ad4fe7e38e2d3a802

manugarg reviewed (commented):
Thanks for this. One minor suggestion.

manugarg commented:
@robpickerill Can you remove protobuf and Go change from this PR. Submitting them through:
https://github.com/google/cloudprober/pull/600

manugarg commented:
@robpickerill Any chance you can squash your commits into one? I can try doing it but it will be another pull request / direct commit (with you still as author) again.

robpickerill commented:
> @robpickerill Any chance you can squash your commits into one? I can try doing it but it will be another pull request / direct commit (with you still as author) again.

I squashed and reworded the commit message for the cleaner commit history - apologies but now fixed
